### PR TITLE
[Feature] Support TIMMBackbone

### PR DESCRIPTION
**Pull request by Junjun2016**
## Motivation

TIMM (https://github.com/rwightman/pytorch-image-models) contains a lot of backbones, which are not available in this repository. But mmsegmentation code is more flexible for training and adding new components for semantic segmentation. It would be convenient to have a wrapper here to use TIMM backbones.

Related PR：https://github.com/open-mmlab/mmclassification/pull/427

## Modification

This PR adds `TIMMBackbone` wrapper to use timm backbones.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
No

## Use cases (Optional)

To use `resnet18` from timm with pretrained weights and output stride 8, you need to use the following backbone config:
```python
backbone=dict(
    type="TIMMBackbone",
    model_name="resnet18",
    features_only=True,
    pretrained=True,   
    output_stride=8)
```
## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by xvjiarui — APPROVED** [review]
#1004 is merge. Maybe we could try if TIMM works with 1.5.0

**Comment by Junjun2016** [comment]
image

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -71,9 +71,17 @@ jobs:
         run: rm -rf .eggs && pip install -e .
       - name: Run unittests and generate coverage report
         run: |
+          pip install timm
           coverage run --branch --source mmseg -m pytest tests/
           coverage xml
           coverage report -m
+        if: ${{matrix.torch >= '1.5.0'}}
```
Comment (xvjiarui) [inline .github/workflows/build.yml:78]: Maybe we could undo the skipping

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -71,9 +71,17 @@ jobs:
...
+      - name: Skip timm unittests and generate coverage report
+        run: |
+          coverage run --branch --source mmseg -m pytest tests/ --ignore tests/test_models/test_backbones/test_timm_backbone.py
```
Comment (xvjiarui) [inline .github/workflows/build.yml:81]: Maybe we could undo the ignore

Comment (Junjun2016) [reply .github/workflows/build.yml:81]: TIMM does not support pt1.3

Comment (Junjun2016) [reply .github/workflows/build.yml:81]: `if: ${{matrix.torch < '1.5.0'}}` will skip the timm unittests.

**Pull request merged**